solr/core/src/java/org/apache/solr/response/JacksonJsonWriter.java

```diff
@@ -43,17 +47,42 @@ public JacksonJsonWriter() {
     jsonfactory = new JsonFactory();
   }
 
+  // let's also implement the binary version since Jackson supports that (probably faster)
   @Override
-  public void write(OutputStream out, SolrQueryRequest request, SolrQueryResponse response)
+  public void write(
+      OutputStream out, SolrQueryRequest request, SolrQueryResponse response, String contentType)
       throws IOException {
-    WriterImpl sw = new WriterImpl(jsonfactory, out, request, response);
+    out = new NonFlushingStream(out);
+    // resolve the encoding
```

Comment:
this charSet check logic is new.  Since Jackson can specify the encoding and since the contentType (with possibly the encoding) is now passed in (new), made sense to choose the right one.